src/CMakeLists.txt

```diff
@@ -45,53 +40,71 @@ SET(common_src
     SocketBuffer.c
     Heap.c
     LinkedList.c
-    )
+)
 
-IF (CMAKE_SYSTEM_NAME MATCHES "Windows")
-    SET(LIBS_SYSTEM ws2_32)
-ELSEIF (CMAKE_SYSTEM_NAME MATCHES "Linux")
-    SET(LIBS_SYSTEM dl)
-ENDIF()
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+    set(LIBS_SYSTEM ws2_32)
+elseif (CMAKE_SYSTEM_NAME MATCHES "Linux")
+    set(LIBS_SYSTEM dl)
+endif()
 
-ADD_EXECUTABLE(MQTTVersion MQTTVersion.c)
-ADD_LIBRARY(paho-mqtt3c SHARED ${common_src} MQTTClient.c)
-ADD_LIBRARY(paho-mqtt3a SHARED ${common_src} MQTTAsync.c)
-TARGET_LINK_LIBRARIES(paho-mqtt3c pthread ${LIBS_SYSTEM}) 
-TARGET_LINK_LIBRARIES(paho-mqtt3a pthread ${LIBS_SYSTEM})
-TARGET_LINK_LIBRARIES(MQTTVersion paho-mqtt3a paho-mqtt3c ${LIBS_SYSTEM})
-SET_TARGET_PROPERTIES(
+add_executable(MQTTVersion MQTTVersion.c)
+add_library(paho-mqtt3c SHARED ${common_src} MQTTClient.c)
+add_library(paho-mqtt3a SHARED ${common_src} MQTTAsync.c)
+target_include_directories(paho-mqtt3c PUBLIC
```

Comment:
You might want to remove those "SHARED" options and use the global BUILD_SHARED_LIBS mechanism, maybe also as an option to set it to shared by default. This way the user can decide if she wants to build static or shared.